### ISSUE TITLE
📎 Implement `useObjectShorthand`

### ISSUE THREAD

**Issue by arendjr** [biomejs/biome#4816]
### Description

Port the [object-shorthand](https://eslint.org/docs/latest/rules/object-shorthand/) rule from eslint.

**Comment by scarf005** [comment]
Hi, I'm interested on opening a PR.

**Comment by ematipico** [comment]
@scarf005 I assigned the issue to you

**Comment by dy0gu** [comment]
Hi there @scarf005, sorry to bother but do you have any updates on this? I also accept pointers on how to get started doing it myself if the implementation is not too complex!

[object-shorthand](https://eslint.org/docs/latest/rules/object-shorthand/) and [no-floating-promisses](https://typescript-eslint.io/rules/no-floating-promises/) are 2 of the rules I miss the most in the midst of all the awesomeness Biome provides.

Since I'm not getting the latter (blocked by #3187), I'd love to help speed this one up just for me personally! 😁

**Comment by AlonMiz** [comment]
this would be great...

**Comment by dyc3** [comment]
Since we haven't heard from @scarf005 in a while, I'll go ahead and reassign this to @dy0gu.

**Comment by dy0gu** [comment]
Hi again, I was wondering if this has to be an exact port of the eslint rule? One team I work with prefers using the old syntax everywhere, instead of the newer ECMAScript 6 form.

Do you think it would make sense to make it `useConsistentObjectDefinition` instead? With a `syntax` option for "ECMAS6" and "Legacy" (names pending patent)...?

**Comment by ematipico** [comment]
Probably better to use a "neutral" name (second suggestion). You could follow the same naming convention of https://biomejs.dev/linter/rules/use-consistent-array-type/#options

**Comment by scarf005** [comment]
is the code action implemented too?

**Comment by ematipico** [comment]
Looks like it's not, I will open a new issue for that

**Comment by minht11** [comment]
I wonder about default value for this rule. Right now rule uses "explicit" syntax, but given that:
- Eslint equivalent rule has default option set to always use shorthand.
- Popular Eslint configs such as [Airbnb](https://github.com/search?q=repo%3Aairbnb%2Fjavascript+object-shorthand&type=code) and  [Antfu](https://github.com/search?q=repo%3Aantfu%2Feslint-config%20object-shorthand&type=code) also use shorthand by default.

Should default option be switched from "explicit" to "shorthand", because thats what most users would expect?

**Comment by AlonMiz** [comment]
> I wonder about default value for this rule. Right now rule uses "explicit" syntax, but given that:
> 
> * Eslint equivalent rule has default option set to always use shorthand.
> * Popular Eslint configs such as [Airbnb](https://github.com/search?q=repo%3Aairbnb%2Fjavascript+object-shorthand&type=code) and  [Antfu](https://github.com/search?q=repo%3Aantfu%2Feslint-config%20object-shorthand&type=code) also use shorthand by default.
> 
> Should default option be switched from "explicit" to "shorthand", because thats what most users would expect?

Yes that's very common. It's even annoying that it's not enabled by default
